File: 백준/Gold/2565. 전깃줄/전깃줄.py
n = int(input())
line = [0] * (n)
for _ in range(n):
    line[_] = list(map(int,input().split()))
line.sort()
# print(line)
check = []
for _ in range(n):
    check.append(line[_][1])
# print(check)
dp = [1 for i in range(n)]
for i in range(n):
    for j in range(i):
        if check[i] > check[j]:
            dp[i] = max(dp[i], dp[j] + 1)
# print(dp)
print(n-max(dp))

File: 백준/Gold/2565. 전깃줄/전깃줄.java
import org.w3c.dom.xpath.XPathResult;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Comparator;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int num = Integer.parseInt(br.readLine());
        int[][] arr = new int[num][];
        int[] dp = new int[num];
        Arrays.fill(dp,1);
        for(int i = 0; i<num;i++){
            arr[i] = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        }
        Arrays.sort(arr, Comparator.comparingInt(a -> a[0]));

        for(int i = 0;i<num ; i++) {
            for (int j = 0; j < i; j++) {
                if (arr[j][1] < arr[i][1]) {
                    dp[i] = Math.max(dp[j] + 1,dp[i]);
                }
            }
        }
        int max = 0;
        for(int k : dp){
            max = max<k ? k:max;
        }

        System.out.println(num - max);
    }
}
